bundles/org.eclipse.ui.workbench/eclipseui/org/eclipse/ui/internal/WindowsDefenderConfigurator.java

```diff
@@ -342,14 +342,12 @@ public static String createAddExclusionsPowershellCommand(String extraSeparator)
 		//
 		// For .NET's stream API called LINQ see:
 		// https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable
-		String excludedPaths = paths.stream().map(Path::toString).map(p -> '"' + p + '"')
-				.collect(Collectors.joining(',' + extraSeparator));
-		final String exclusionType = "ExclusionProcess"; //$NON-NLS-1$
-		return String.join(';' + extraSeparator, "$exclusions=@(" + extraSeparator + excludedPaths + ')', //$NON-NLS-1$
-				"$existingExclusions=[Collections.Generic.HashSet[String]](Get-MpPreference)." + exclusionType, //$NON-NLS-1$
-				"if($existingExclusions -eq $null) { $existingExclusions = New-Object Collections.Generic.HashSet[String] }", //$NON-NLS-1$
-				"$exclusionsToAdd=[Linq.Enumerable]::ToArray([Linq.Enumerable]::Where($exclusions,[Func[object,bool]]{param($ex)!$existingExclusions.Contains($ex)}))", //$NON-NLS-1$
-				"if($exclusionsToAdd.Length -gt 0){ Add-MpPreference -" + exclusionType + " $exclusionsToAdd }"); //$NON-NLS-1$ //$NON-NLS-2$
+		String excludedPaths = paths.stream().map(Path::toString).map(p -> "'" + p.replace("'", "''") + "'")
+				.collect(Collectors.joining(','));
+		return String.join(";",
+				"$pathsToExclude = @(" + excludedPaths + ")",
+				"Add-MpPreference -ExclusionProcess $pathsToExclude"
```

Comment:
Please keep the `extraSeparator`s here as well and keep the name of the name of the variable:
```suggestion
		return String.join(';' + extraSeparator, "$exclusions=@(" + extraSeparator + excludedPaths + ')', //$NON-NLS-1$
				"Add-MpPreference -ExclusionProcess $pathsToExclude"
```
Alternatively, as the script seems now easy enough, we could skip the creation of a variable and just pass the string-array directly to make it even more compact:
```suggestion
		return "Add-MpPreference -ExclusionProcess " + extraSeparator + excludedPaths; //$NON-NLS-1$
```